internal/signerverifier/sigstore/sigstore.go

```diff
@@ -175,80 +163,29 @@ func (v *Verifier) ExpectedExtensionKind() string {
 	return ExtensionMimeType
 }
 
-func (v *Verifier) getTUFRoot() (root.TrustedMaterial, bool, error) {
-	// The env vars we look at for private sigstore:
-	// SIGSTORE_ROOT_FILE -> the Fulcio root
-	// SIGSTORE_CT_LOG_PUBLIC_KEY_FILE -> Fulcio's CT Log pubkey
-	// SIGSTORE_REKOR_PUBLIC_KEY -> Rekor's pubkey
-	// TODO: Support ctlog and tsa
-	fulcioRootFilePath := os.Getenv(EnvSigstoreRootFile)
-	ctLogPublicKeyFilePath := os.Getenv(EnvSigstoreCTLogPublicKeyFile)
-	rekorPublicKeyFilePath := os.Getenv(EnvSigstoreRekorPublicKey)
-
-	if fulcioRootFilePath != "" || ctLogPublicKeyFilePath != "" || rekorPublicKeyFilePath != "" {
-		// if any env var is set, require all?
-		if fulcioRootFilePath == "" || ctLogPublicKeyFilePath == "" || rekorPublicKeyFilePath == "" {
-			return nil, false, fmt.Errorf("partial env var set") // TODO
-		}
-
-		slog.Debug("Using environment variables to establish trust for Sigstore instance...")
-
-		fulcioCertAuthorities := []root.CertificateAuthority{}
-		cert, err := parsePEMFile(fulcioRootFilePath)
-		if err != nil {
-			return nil, false, err
-		}
-		fulcioCertAuthorities = append(fulcioCertAuthorities, *cert)
-
-		rekorPubKeyBytes, err := os.ReadFile(rekorPublicKeyFilePath)
-		if err != nil {
-			return nil, false, err
-		}
-		block, _ := pem.Decode(rekorPubKeyBytes)
-		if block == nil {
-			return nil, false, fmt.Errorf("failed to decode rekor public key")
-		}
-		rekorKey, err := x509.ParsePKIXPublicKey(block.Bytes)
+func (v *Verifier) getTUFRoot() (root.TrustedMaterial, error) {
+	rootFilePath := os.Getenv("SIGSTORE_TRUSTED_ROOT")
+	if rootFilePath == "" {
+		slog.Debug("SIGSTORE_TRUSTED_ROOT not set, using public good instance")
+		opts := sigstoretuf.DefaultOptions()
+		c, err := sigstoretuf.New(opts)
 		if err != nil {
-			return nil, false, err
+			return nil, err
 		}
 
-		keyHash := sha256.Sum256(block.Bytes)
-		keyID := hex.EncodeToString(keyHash[:])
-
-		rekorTransparencyLog := &root.TransparencyLog{
-			BaseURL:           v.rekorURL,
-			HashFunc:          crypto.SHA256,
-			ID:                keyHash[:],
-			PublicKey:         rekorKey,
-			SignatureHashFunc: crypto.SHA256,
-		}
-		rekorTransparencyLogs := map[string]*root.TransparencyLog{
-			keyID: rekorTransparencyLog,
-		}
-
-		// TODO: CT Log
-		// TODO TSA
-
-		trustedRoot, err := root.NewTrustedRoot(root.TrustedRootMediaType01, fulcioCertAuthorities, nil, nil, rekorTransparencyLogs)
-		return trustedRoot, true, err
+		return root.GetTrustedRoot(c)
 	}
 
-	// Use the TUF flow
-	// TODO: support custom sigstore TUF root URL
-
-	tufClient, err := sigstoretuf.New(sigstoretuf.DefaultOptions())
+	trustedRootBytes, err := os.ReadFile(rootFilePath)
```

Comment:
This is a bit off. Unfortunately, "root" is a bit overloaded here. We want to load the root.json bytes for the TUF repository, that's what's embedded in the sigstore-go library as well for the public good instance. Then, using that TUF root and the TUF client, we want to separately fetch `trusted_root.json` which is not the same as the TUF root.

TUF root -> https://github.com/sigstore/sigstore-go/blob/main/pkg/tuf/repository/root.json
trusted_root.json -> https://tuf-repo-cdn.sigstore.dev/targets/f44a1b88128e55ebfb62189becbc0fa48d4ec9915c65ac54ba0e46a008b12d5b.trusted_root.json on the public good instance

Reply:
I suggest you load the TUF root from the env var if it's set. Then override the tuf client's options to set the bytes you loaded instead of the default, embedded tuf root file.